### PR TITLE
support python 3.11

### PR DESCRIPTION
**Pull request by wintonzheng**
The main goal is to make wyvern compatible with python `3.11 (^3.8,<3.12)`
- httpx doesn't work with fastapi + python3.11 somehow (i didn't figure out why. but since we already had a plan to migrate to aiohttp as it's faster than httpx, i just did the migration from httpx to aiohttp (`wyvern.core.httpx -> wyvern.core.http`))
- removed some unnecessary info log and fix some warning log in text
- introduced aioresponses as a dev dependency to help mock aiohttp response

TODOs:
- [x] test WyvernAPI 
- [x] test with Wyvern Pipeline service

- [ ] Does this PR have impact on local development experience? If yes, make sure you have a plan and add the documentations to address issues that come with the change
- [ ] bump version
- [ ] make a release
- [ ] publish to pypi service

### CODE REVIEW

File: tests/feature_store/test_real_time_features.py [hunk 1]
```diff
@@ -18,21 +18,19 @@ def mock_redis(mocker):
     """
     Mocks the redis call. Each entry under `return_value` corresponds to a single entity fetch from Redis
     """
-    with mocker.patch(
```
Comment (wintonzheng) [inline tests/feature_store/test_real_time_features.py:21]: this is to fix:
```
PytestMockWarning: Mocks returned by pytest-mock do not need to be used as context managers. The mocker fixture automatically undoes mocking at the end of a test. This warning can be ignored if it was triggered by mocking a context manager. https://pytest-mock.readthedocs.io/en/latest/remarks.html#usage-as-context-manager
    with mocker.patch(
```

**Review by suchintan — APPROVED** [review]
This one seems like a pretty big change! Excited to support 3.11

File: wyvern/core/http.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    def __call__(self):
+        """Calling the instantiated AiohttpClientWrapper returns the wrapped singleton."""
+        # Ensure we don't use it if not started / running
+        if self.async_client is None:
+            raise WyvernError("AiohttpClientWrapper not started")
+
+        return self.async_client
+
+
+aiohttp_client = AiohttpClientWrapper()
```
Comment (suchintan) [inline wyvern/core/http.py:38]: Should we just call this http_client? Or does it make sense to also keep the aio

Comment (wintonzheng) [reply wyvern/core/http.py:38]: yeah, good idea. i'll update

Comment (wintonzheng) [reply wyvern/core/http.py:38]: actually I'll keep it aiohttp_client for now.

**Pull request merged**